tests/validators/test_dict.py

```diff
@@ -157,17 +163,68 @@ def __len__(self):
 
     assert exc_info.value.errors(include_url=False) == [
         {
-            'type': 'mapping_type',
+            'type': 'dict_type',
             'loc': (),
-            'msg': 'Input should be a valid mapping, error: RuntimeError: intentional error',
+            'msg': 'Input should be a valid dictionary',
             'input': HasRepr(IsStr(regex='.+BadMapping object at.+')),
-            'ctx': {'error': 'RuntimeError: intentional error'},
         }
     ]
 
 
-@pytest.mark.parametrize('mapping_items', [[(1,)], ['foobar'], [(1, 2, 3)], 'not list'])
-def test_mapping_error_yield_1(mapping_items):
+@pytest.mark.parametrize(
+    'mapping_items,errors',
+    [
+        (
+            [(1,)],
+            [
+                {
+                    'type': 'iteration_error',
+                    'loc': (0,),
+                    'msg': 'Error iterating over object, error: ValueError: expected tuple of length 2, but got tuple of length 1',  # noqa: E501
+                    'input': HasRepr(IsStr(regex='.+BadMapping object at.+')),
+                    'ctx': {'error': 'ValueError: expected tuple of length 2, but got tuple of length 1'},
+                }
+            ],
+        ),
```

Comment:
`pytest.param` allows to have less braces and makes `pytest.mark.parametrize` arguments more readable. Also, explicit ids help understanding cases better.